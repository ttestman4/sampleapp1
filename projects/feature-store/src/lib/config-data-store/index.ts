export * from './config-data-store.module';
