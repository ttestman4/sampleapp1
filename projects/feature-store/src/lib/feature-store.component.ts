import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'lib-feature-store',
  template: `
    <p>
      feature-store works!
    </p>
  `,
  styles: []
})
export class FeatureStoreComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
