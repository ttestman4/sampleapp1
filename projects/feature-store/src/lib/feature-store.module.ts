import { NgModule } from '@angular/core';
import { FeatureStoreComponent } from './feature-store.component';

@NgModule({
  declarations: [FeatureStoreComponent],
  imports: [
  ],
  exports: [FeatureStoreComponent]
})
export class FeatureStoreModule { }
