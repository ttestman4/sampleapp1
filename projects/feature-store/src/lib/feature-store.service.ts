import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class FeatureStoreService {

  constructor() { }
}
