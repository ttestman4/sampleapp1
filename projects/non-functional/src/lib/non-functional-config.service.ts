import { InjectionToken } from '@angular/core';
import { NonFunctionalConfig } from './non-functional.models';

/**
 * This is not a real service, but it looks like it from the outside.
 * It's just an InjectionToken used to import the config object,
 * provided from the outside
 */
export const NonFunctionalConfigService =
    new InjectionToken<NonFunctionalConfig>('NonFunctionalConfig');
