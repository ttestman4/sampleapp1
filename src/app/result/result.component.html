<mat-accordion>
  <mat-expansion-panel *ngFor="let flightCriteria of flightCriteriaDetails; let i= index" [expanded]="i === 0">
    <mat-expansion-panel-header>
      <mat-panel-title>
        <mat-icon mat-list-icon>{{flightCriteria.travelOrder === 1 ? 'flight_land': 'flight_takeoff'}}</mat-icon>
        {{flightCriteria.origin}} to {{flightCriteria.destination}}
      </mat-panel-title>
    </mat-expansion-panel-header>
    <mat-accordion>
      <mat-expansion-panel *ngFor="let flight of flightListAll[i]">
        <mat-expansion-panel-header>
          <mat-panel-title>{{flight.name}}
            {{flight.departureTime | time}}
            {{flight.arrivalTime | time }}
            {{flight.duration | duration }}
            {{flight.price | currency: 'INR' : 'symbol': '.0-0'}}</mat-panel-title>
        </mat-expansion-panel-header>
        <div *ngFor="let flightLeg of flight.multiple">
          {{flightLeg.name}}
          {{flightLeg.departureTime | time}}
          {{flightLeg.arrivalTime | time }}
          {{flightLeg.duration | duration }}
          {{flightLeg.price | currency: 'INR' : 'symbol': '.0-0'}}
        </div>
        <div *ngIf="flight.name != 'Multiple'">
          {{flight.name}}
          {{flight.departureTime | time}}
          {{flight.arrivalTime | time }}
          {{flight.duration | duration }}
          {{flight.price | currency: 'INR' : 'symbol': '.0-0'}}
        </div>
      </mat-expansion-panel>
    </mat-accordion>
  </mat-expansion-panel>
</mat-accordion>
