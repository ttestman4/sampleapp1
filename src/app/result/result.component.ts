import { Component, OnDestroy, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import * as FeatuerStore from 'feature-store';
import { Subscription } from 'rxjs';

@Component({
  selector: 'spa-result',
  templateUrl: './result.component.html',
  styleUrls: ['./result.component.scss']
})
export class ResultComponent implements OnInit, OnDestroy {
  flightResultSubscription: Subscription;
  flightCriteriaFlightDetailsSubscription: Subscription;
  flightListAll: FeatuerStore.FlightResultDetail[][] = [];
  flightCriteriaDetails: FeatuerStore.FlightSearchDetails[] = [];
  constructor(private store: Store<FeatuerStore.SearchState>) {
    this.flightResultSubscription = this.store.pipe(
      select(FeatuerStore.selectResultFlightDetailsListAllFilteredByPrice)
    ).subscribe((flightDetails) => {
      this.flightListAll = flightDetails;
    });

    this.flightCriteriaFlightDetailsSubscription = this.store.pipe(
      select(FeatuerStore.selectCriteriaFlightDetails)
    ).subscribe((flightDetails) => {
      this.flightCriteriaDetails = flightDetails;
    });
  }

  ngOnInit() {
  }

  ngOnDestroy() {
    this.flightResultSubscription.unsubscribe();
    this.flightCriteriaFlightDetailsSubscription.unsubscribe();
  }
}
