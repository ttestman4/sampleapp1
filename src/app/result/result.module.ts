import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ResultComponent } from './result.component';

@NgModule({
  declarations: [ResultComponent],
  imports: [
    CommonModule
  ],
  exports: [
    ResultComponent
  ]
})
export class ResultModule { }
