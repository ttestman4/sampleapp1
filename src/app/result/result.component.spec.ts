import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { FlightStoreModule } from 'feature-store';
import { CustomLoggerLevel, NonFunctionalModule } from 'non-functional';
import { ConfigureFn, configureTests } from '../../lib/testing';
import { ResultComponent } from './result.component';


describe('ResultComponent', () => {
  let component: ResultComponent;
  let fixture: ComponentFixture<ResultComponent>;

  beforeEach(async(() => {
    NonFunctionalModule.forTestReset();
    FlightStoreModule.forTestReset();
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [ResultComponent],
        imports: [
          NoopAnimationsModule,
          FormsModule,
          ReactiveFormsModule,
          NonFunctionalModule.forRoot({
            serverLoggingUrl: '/api/logs',
            level: CustomLoggerLevel.DEBUG,
            serverLogLevel: CustomLoggerLevel.ERROR,
            disableConsoleLogging: true,
            debugStore: false
          }),
          // ConfigDataStoreModule.forChild(),
          FlightStoreModule.forRoot(),
          RouterTestingModule.withRoutes([]),
        ],
        schemas: [NO_ERRORS_SCHEMA],
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(ResultComponent);
      component = fixture.componentInstance;
      fixture.detectChanges();
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snaps', () => {
    expect(fixture).toMatchSnapshot();
  });
});
