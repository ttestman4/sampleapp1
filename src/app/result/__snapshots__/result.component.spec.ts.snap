// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultComponent snaps 1`] = `
<spa-result
  flightCriteriaDetails={[Function Array]}
  flightCriteriaFlightDetailsSubscription={[Function Subscriber]}
  flightListAll={[Function Array]}
  flightResultSubscription={[Function Subscriber]}
  store={[Function Store]}
>
  <mat-accordion>
    
  </mat-accordion>
</spa-result>
`;
